packages/@lwc/compiler/src/transformers/template.ts

```diff
@@ -53,14 +53,22 @@ export default function templateTransform(
     };
 }
 
+function escapeScopeToken(input: string) {
+    // Minimal escape for strings containing the "@" character, which is disallowed in CSS selectors
+    // and at the beginning of attribute names
+    return input.replace(/@/g, '___at___');
```

Comment:
The `@` character is not a valid beginning of attribute character, but it's also not a valid beginning for a custom element name. One of the reasons why I was reluctant to fix this CSS attribute escaping issue is that namespaces starting with a non-alpha character can't be addressed via the template as it is not a valid HTML namespace.

If we take your example below how do reference the `@x/component` via the template?

FYI @nolanlawson @abdulsattar

Reply:
@pmdartus In order to repro this issue (see the test), I used the tag name `<x-component>` but the folder structure `@x/component`. I assume LWR is doing something like this in order to reach this state.

I was hesitant about fixing this too, but it ended up being a simple string replacement that only affects the synthetic shadow scope token, so it seemed low-risk to me.

Reply:
I agree with the fix itself, but I think this doesn't address the underlying issue.